src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +560,242 @@ class RigidBodyComponentSystem extends ComponentSystem {
+     * @param {number} axis - The local space axis with which the capsule's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} position - The world space position for the capsule to be.
+     * @param {Vec3} rotation - The world space rotation for the capsule to have.
+     *
+     * @returns {RaycastResult[]} An array of capsulecast hit results (0 length if there were no hits).
+     */
+    capsuleCast(radius, height, axis, position, rotation) {
+        Debug.assert(Ammo.ConcreteContactResultCallback, 'pc.RigidBodyComponentSystem#capsuleCast: Your version of ammo.js does not expose Ammo.ConcreteContactResultCallback. Update it to latest.');
+
+        if (axis === 0) {
+            axis = 'btCapsuleShapeX';
+        } else if (axis === 2) {
+            axis = 'btCapsuleShapeZ';
+        } else {
+            axis = 'btCapsuleShape';
+        }
+
+        return this._shapecast(new Ammo[axis](radius, height), position, rotation);
+    }
+
+    /**
+     * Perform a collision check on the world and return all entities the cone hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {number} radius - The radius of the cone.
+     * @param {number} height - The total height of the cone from tip to tip.
+     * @param {number} axis - The local space axis with which the cone's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} position - The world space position for the cone to be.
+     * @param {Vec3} rotation - The world space rotation for the cone to have.
+     *
+     * @returns {RaycastResult[]} An array of conecast hit results (0 length if there were no hits).
+     */
+    coneCast(radius, height, axis, position, rotation) {
+        Debug.assert(Ammo.ConcreteContactResultCallback, 'pc.RigidBodyComponentSystem#coneCast: Your version of ammo.js does not expose Ammo.ConcreteContactResultCallback. Update it to latest.');
+
+        if (axis === 0) {
+            axis = 'btConeShapeX';
+        } else if (axis === 2) {
+            axis = 'btConeShapeZ';
+        } else {
+            axis = 'btConeShape';
+        }
+
+        return this._shapecast(new Ammo[axis](radius, height), position, rotation);
+    }
+
+    /**
+     * Perform a collision check on the world and return all entities the cylinder hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {number} radius - The radius of the cylinder.
+     * @param {number} height - The total height of the cylinder from tip to tip.
+     * @param {number} axis - The local space axis with which the cylinder's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} position - The world space position for the cylinder to be.
```

Comment:
it seems both position and rotation is optional, so use []. For example 
```
@param {Vec3} [position] - The world ...
```

The question is - should at least position be required?